src/components/modules/blockEvents.ts

```diff
@@ -332,17 +336,84 @@ export default class BlockEvents extends Module {
       /**
        * Merge Blocks
        */
-      this.mergeBlocks();
+      this.mergeBlocks(BlockManager.previousBlock, BlockManager.currentBlock);
+    }
+  }
+
+  /**
+   * Handle delete keydown on Block
+   *
+   * @param {KeyboardEvent} event - keydown
+   */
+  private deletePressed(event: KeyboardEvent): void {
+    const { BlockManager, BlockSelection, Caret } = this.Editor;
+    const currentBlock = BlockManager.currentBlock;
+    const tool = currentBlock.tool;
+
+    /**
+     * Check if Block should be removed by current Delete keydown
+     */
+    if (currentBlock.selected || (currentBlock.isEmpty && currentBlock.currentInput === currentBlock.firstInput)) {
```

Comment:
The last condition is not clear for me. I suppose there should be `currentBlock.lastInput` instead
But I believe such behaviour is not really intuitive, let me explain why.
For example you have list tool, you create couple of lines (but don't type any text there) and press delete key on the last line. You wouldn't expect this will delete the list block

Reply:
Well, it makes sense for paragraph tool, but for more complex tools it looks unintuitive for me. When I press delete I expect next entity to be deleted (or merged with the current one), but not current entity deletion. I might be wrong, so could you please apply changes and I'll take a look in action